### ISSUE TITLE
Bug: panic inside range

### ISSUE THREAD

**Issue by the-plash** [risor-io/risor#211]
The following risor code panics when run:
```go
for i := range 1 {
  try(func() {
    print("hello")
    error("kaboom")
  })
}
```
```
hello
panic: interface conversion: *object.NilType is not object.Iterator: missing method Entry
```

Assigning the return value of `print(...)` to a local variable fixes the problem

```go
for i := range 1 {
  try(func() {
    x := print("hello")
    error("kaboom")
  })
}
```
`hello`

The bug seems to be related to the implicit nil return value from `print(...)` which pollutes the stack. The same behaviour is observed more clearly with the following code:
```go
for i := range 1 {
  try(func() {
    "hello"
    error("kaboom")
  })
}
```
`panic: interface conversion: *object.String is not object.Iterator: missing method Entry`

The string `"hello"` is implicitly returned and is popped from the stack instead of the expected `IntIter` when resuming. Unfortunately i don't have any more insight on why this may be happening.

Possibly related to #173

**Comment by myzie** [comment]
Thanks for the thorough bug report @the-plash. I'll track it down.

**Comment by myzie** [comment]
Hi @the-plash, this should be fixed in v1.5.2. Let me know how it does for you.
